@xen-orchestra/xapi/src/vbd.js

```diff
@@ -66,4 +67,21 @@ module.exports = class Vbd {
       await this.callAsync('VBD.plug', vbdRef)
     }
   }
+
+  async disconnect(ref) {
+    // TODO: check if VBD is attached before
+    try {
+      await this.call('VBD.unplug_force', ref)
+    } catch (error) {
+      if (error.code === 'VBD_NOT_UNPLUGGABLE') {
+        await this.setField('VBD', ref, 'unpluggable', true)
+        await this.call('VBD.unplug_force', ref)
+      }
+    }
```

Comment:
This code is copied from https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/xapi/index.js#L1791-L1801.

IMHO, we shouldn't ignore silently the call error. We can at least log the error.

Reply:
Simply throw it.